### ISSUE TITLE
Unused import causes linker failure with -betterC

### ISSUE THREAD

**Issue by andy-hanson** [ldc-developers/ldc#3380]
System: openSUSE Tumbleweed
Architecture: x86_64
LLVM: 9.0.1 (from distro packages)
LDC: ldc2-1.21.0-beta1-linux-x86_64

Can be reproduced with just:

```
import std.stdio : writeln;
extern(C) int main() {
	return "foo" == "bar";
}
```

`dmd -betterC app.d` succeeds.

`ldc2 -betterC app.d` fails with:

```
app.o:app.d:function main: error: undefined reference to '_D4core8internal5array8equality__T8__equalsTyaTyaZQqFNaNbNiNfAyaQdZb'
collect2: error: ld returned 1 exit status
Error: /usr/bin/cc failed with status: 1
```

There is no error if the unused import is removed.

**Comment by kinke** [comment]
Please retry with `-allinst`.

**Comment by andy-hanson** [comment]
`-allinst` fixes it.

**Comment by JohanEngelen** [comment]
> This has to do with template culling and is extremely brittle (especially with `-betterC`)

When `-betterC` is passed, we could disable all template culling for all druntime/phobos symbols that have druntime/phobos as root?

**Comment by kinke** [comment]
I suppose that could be handled in the frontend, but frankly, I wouldn't be thrilled about more `-betterC` hacks and workarounds. Even regular druntime/Phobos contributors don't all know that one of the -betterC implications is that druntime and Phobos aren't linked...

**Comment by rikkimax** [comment]
This should have been fixed upstream, works as of 1.39.0, can be closed.
